smarts/sstudio/genhistories.py

```diff
@@ -1032,24 +1093,25 @@ def _check_args(args) -> bool:
         os.remove(args.output)
 
     if args.old:
-        dataset_spec = {"source": "OldJSON", "input_path": args.dataset}
+        dataset_spec = {"source_type": "OldJSON", "input_path": args.dataset}
     else:
         with open(args.dataset, "r") as yf:
             dataset_spec = yaml.safe_load(yf)["trajectory_dataset"]
 
-    default_input_path = "<PATH TO FILE GOES HERE>"
-    if dataset_spec.get("input_path") == default_input_path:
+    if not dataset_spec.get("input_path"):
         print(f"skipping placeholder dataset spec at {args.dataset}.")
         sys.exit(0)
 
+    _migrate_deprecated_dataset(dataset_spec)
+
     if args.x_offset:
         dataset_spec["x_offset"] = args.x_offset
 
     if args.y_offset:
         dataset_spec["y_offset"] = args.y_offset
 
-    source = dataset_spec.get("source", "NGSIM")
-    if source == "NGSIM" or source == "NGSIM2":
+    source = dataset_spec.get("source_type", "NGSIM_highway")
+    if source.startswith("NGSIM"):
         dataset = NGSIM(dataset_spec, args.output)
     elif source == "Waymo":
         dataset = Waymo(dataset_spec, args.output)
```

Comment:
This and the rest of the lines seem redundant after the addition of `import_dataset` function. Maybe `import_dataset` should be modified to allow passing in both,  `TrafficHistoryDataset` object and user arguments.

Reply:
Yeah, it is redundant, I agree.  I mainly left it "as is" to continue to support the deprecated yaml-file approach.  But when we remove that, I had planned to change the `__main__` path to just call `import_dataset()` so that `gehistories.py` can continue to be used either as a script or as a source-code file.